.github/workflows/release.yml

```diff
@@ -45,19 +45,63 @@ jobs:
         name: pganalyze-collector-linux-amd64
         path: pganalyze-collector-linux-amd64
 
+  build_packages:
+    # Use older Ubuntu release to ensure availability of CGroupsv1 (which are necessary
+    # currently to support older systemd on some of the test containers)
+    runs-on: ubuntu-20.04
+    permissions: {}
+    # ensure we don't release on create events for branches (only tags)
+    if: ${{ startsWith( github.ref, 'refs/tags' ) }}
+
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+
+    - name: Build packages
+      run: |
+        DOCKER_BUILDKIT=1 make -C packages build
+
+    - name: Test packages
+      run: |
+        DOCKER_BUILDKIT=1 make -C packages test
+
+    - name: Upload packages as artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: packages
+        path: |
+          packages/tmp/pganalyze-collector-*.aarch64.rpm
+          packages/tmp/pganalyze-collector-*.x86_64.rpm
+          packages/tmp/pganalyze-collector_*_amd64.deb
+          packages/tmp/pganalyze-collector_*_arm64.deb
+        if-no-files-found: error
+
   release:
-    needs: build
+    needs: [build, build_packages]
     runs-on: ubuntu-latest
+    # ensure we don't release on create events for branches (only tags)
+    if: ${{ startsWith( github.ref, 'refs/tags' ) }}
 
     steps:
     - name: Download build
-      uses: actions/download-artifact@v2
+      uses: actions/download-artifact@v3
       with:
         name: pganalyze-collector-linux-amd64
 
+    - name: Download build packages
+      uses: actions/download-artifact@v3
+      with:
+        name: packages
+
     - name: Get version from tag
       id: get_version
-      run: echo ::set-output name=version::${GITHUB_REF#refs/tags/}
+      run: echo "version=${GITHUB_REF#refs/tags/}" >> $GITHUB_OUTPUT
```

Comment:
For what it's worth, we should be able to use GITHUB_REF_NAME here: https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

Reply:
Nice! Since I didn't test with that I'll leave that refactoring out for now, but good idea to clean it up that way.